server/src/test/java/io/crate/expression/scalar/AgeFunctionTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+package io.crate.expression.scalar;
+
+
+import io.crate.metadata.SystemClock;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.LocalDate;
+import java.time.Month;
+import java.time.ZoneOffset;
+
+public class AgeFunctionTest extends ScalarTestCase {
+
+    private static final long FIRST_JAN_2021_MIDNIGHT_UTC_AS_MILLIS =
+        LocalDate.of(2021, Month.JANUARY, 1).atStartOfDay().toInstant(ZoneOffset.UTC).toEpochMilli();
+
+    @Before
+    public void prepare() {
+        SystemClock.setCurrentMillisFixedUTC(FIRST_JAN_2021_MIDNIGHT_UTC_AS_MILLIS);
+    }
+
+    @After
+    public void cleanUp() {
+        SystemClock.setCurrentMillisSystemUTC();
+    }
+
+    @Test
+    public void test_calls_within_statement_are_idempotent() {
+        assertEvaluate("age('2019-01-01' :: TIMESTAMP) = age('2019-01-01' :: TIMESTAMP)", true);
+    }
+
+    @Test
+    public void test_1_argument_before_curdate_age_is_positive_() {
+        // 1 second left to the current_date (manually set 2021.01.01 midnight)
+        assertEvaluate("pg_catalog.age('2020-12-31 23:59:59'::timestamp without time zone) ", 1000L);
+    }
+
+    @Test
+    public void test_1_argument_before_curdate_age_is_negative() {
+        // 1 second after the current_date (manually set 2021.01.01 midnight)
+        assertEvaluate("pg_catalog.age('2021-01-01 00:00:01'::timestamp without time zone) ", -1000L);
+    }
+
+    @Test
+    public void test_floating_point_arguments() {
+        // We treat float and double values as seconds with milliseconds as fractions.
```

Comment:
initially I had concerns that BinaryScalar's sanitize without `*1000` will make this test fail but was proven wrong - `implicitCast` called before sanitizeValue implicitly.